components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.components.feature.session
+
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+
+/**
+ * Feature implementation for handling fullscreen mode (exiting and back button presses).
+ */
+internal open class FullScreenFeature(
```

Comment:
> * Don't think it should be `internal` otherwise apps can't use it?

You're right, I don't need internal!

> * Also not sure why it's `open`? Is it because you wanted to create a spy for it? If so you can add this file instead: https://github.com/mozilla-mobile/android-components/blob/master/components/feature/awesomebar/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker

I wanted to access the `activeSession` attribute in the `SelectionAwareSessionObserver` in my tests which is protected. I don't think the MockMaker is enough to do that right?

Reply:
OK, we could just make `activeSession` internal, I guess, but either way that's fine.

Reply:
That's a good point, I wasn't sure about changing this at the time because I felt like the protected was there for a reason.. which it turns out, is true! 😛 

I can't change protected from internal without it breaking a bunch of other classes since the `SelectionAwareSessionObserver` isn't in the same package as the others.

Reply:
Hm that explains why `activeSession` is `protected`, but not why it's `open`, which seems like an oversight. That's for another time. All fine here :).